src/resources/js/views/events-bar.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Makes sure we have all the required levels on the Tribe Object
+ *
+ * @since TBD
+ *
+ * @type   {PlainObject}
+ */
+tribe.events = tribe.events || {};
+tribe.events.views = tribe.events.views || {};
+
+/**
+ * Configures Events Bar Object in the Global Tribe variable
+ *
+ * @since TBD
+ *
+ * @type   {PlainObject}
+ */
+tribe.events.views.eventsBar = {};
+
+/**
+ * Initializes in a Strict env the code that manages the Event Views
+ *
+ * @since TBD
+ *
+ * @param  {PlainObject} $   jQuery
+ * @param  {PlainObject} obj tribe.events.views.manager
+ *
+ * @return {void}
+ */
+( function( $, obj ) {
+	'use strict';
+	var $document = $( document );
+
+	/**
+	 * Selectors used for configuration and setup
+	 *
+	 * @since TBD
+	 *
+	 * @type {PlainObject}
+	 */
+	obj.selectors = {
+		inputKeyword: '.tribe-common-c-search__input--keyword',
+		inputKeywordFocus: '.tribe-common-c-search__input--keyword-focus',
+		inputLocation: '.tribe-common-c-search__input--location',
+		inputLocationFocus: '.tribe-common-c-search__input--location-focus',
+	};
+
+	/**
+	 * Toggles accordion on header click
+	 *
+	 * @since TBD
+	 *
+	 * @param {Event} event event object of click event
+	 *
+	 * @return {void}
+	 */
+	obj.inputSetFocusMaybe = function( event ) {
+		var $input = $( event.target );
+
+		// Set the focus class if it has content.
+		$input.toggleClass( event.data.inputClassFocus, '' !== $input.val().trim() );
```

Comment:
either use `event.data.target` here (`event.data.target` is a jQuery object already) or don't pass `target` in the `on()` function